src/States/MapViewStateDraw.cpp

```diff
@@ -254,62 +254,37 @@ void MapViewState::drawNavInfo()
 {
 	Renderer& r = Utility<Renderer>::get();
 
-	// Up / Down
-	if (isPointInRect(MOUSE_COORDS, MOVE_DOWN_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_DOWN_ICON.x(), MOVE_DOWN_ICON.y(), 64, 128, 32, 32, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_DOWN_ICON.x(), MOVE_DOWN_ICON.y(), 64, 128, 32, 32);
-	}
-
-	if (isPointInRect(MOUSE_COORDS, MOVE_UP_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_UP_ICON.x(), MOVE_UP_ICON.y(), 96, 128, 32, 32, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_UP_ICON.x(), MOVE_UP_ICON.y(), 96, 128, 32, 32);
-	}
-
-	// East / West / North / South
-	if (isPointInRect(MOUSE_COORDS, MOVE_EAST_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_EAST_ICON.x(), MOVE_EAST_ICON.y(), 32, 128, 32, 16, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_EAST_ICON.x(), MOVE_EAST_ICON.y(), 32, 128, 32, 16);
-	}
-
-	if (isPointInRect(MOUSE_COORDS, MOVE_WEST_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_WEST_ICON.x(), MOVE_WEST_ICON.y(), 32, 144, 32, 16, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_WEST_ICON.x(), MOVE_WEST_ICON.y(), 32, 144, 32, 16);
-	}
-
-	if (isPointInRect(MOUSE_COORDS, MOVE_NORTH_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_NORTH_ICON.x(), MOVE_NORTH_ICON.y(), 0, 128, 32, 16, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_NORTH_ICON.x(), MOVE_NORTH_ICON.y(), 0, 128, 32, 16);
-	}
-
-	if (isPointInRect(MOUSE_COORDS, MOVE_SOUTH_ICON))
-	{
-		r.drawSubImage(mUiIcons, MOVE_SOUTH_ICON.x(), MOVE_SOUTH_ICON.y(), 0, 144, 32, 16, 255, 0, 0, 255);
-	}
-	else
-	{
-		r.drawSubImage(mUiIcons, MOVE_SOUTH_ICON.x(), MOVE_SOUTH_ICON.y(), 0, 144, 32, 16);
-	}
-
+	NAS2D::Rectangle_2d currentIconBounds = MOVE_DOWN_ICON;
+	bool isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
+	NAS2D::Color iconColor{255, 255, 255, 255};
+	NAS2D::Color iconHighlightColor{255, 0, 0, 255};
+	NAS2D::Color color = isMouseInIcon ? iconHighlightColor : iconColor;
+	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 64, 128, 32, 32, color.red(), color.green(), color.blue(), color.alpha());
+
+	currentIconBounds = MOVE_UP_ICON;
+	isMouseInIcon = isPointInRect(MOUSE_COORDS, currentIconBounds);
+	color = isMouseInIcon ? iconHighlightColor : iconColor;
+	r.drawSubImage(mUiIcons, currentIconBounds.x(), currentIconBounds.y(), 96, 128, 32, 32, color.red(), color.green(), color.blue(), color.alpha());
```

Comment:
I like what you're going for here. It seems a bit off though. It's kind of hard to read. There's a lot of duplication. Variables are being re-used in different contexts. This seems like a sign the code can be extracted to a function.

The magic numbers can probably be extracted to `Rectangle` structs. It looks like they are associated with an icon. A function could then take each icon `Point` and associated "magic number" `Rectangle`, as well as the current mouse `Point`, and draw it with any appropriate highlight. Not sure if you want the highlight colors to be parameters too.